src/argilla_server/validators/responses.py

```diff
@@ -12,21 +12,16 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
+from typing import Union
+
 from argilla_server.enums import QuestionType, ResponseStatus
 from argilla_server.models import Record
 from argilla_server.schemas.v1.responses import ResponseCreate, ResponseUpdate, ResponseUpsert
-from argilla_server.validators.response_values import (
-    LabelSelectionQuestionResponseValueValidator,
-    MultiLabelSelectionQuestionResponseValueValidator,
-    RankingQuestionResponseValueValidator,
-    RatingQuestionResponseValueValidator,
-    SpanQuestionResponseValueValidator,
-    TextQuestionResponseValueValidator,
-)
-
-
-class ResponseCreateValidator:
-    def __init__(self, response_change: ResponseCreate):
+from argilla_server.validators.response_values import ResponseValueValidator
+
+
+class ResponseValidator:
+    def __init__(self, response_change: Union[ResponseCreate, ResponseUpdate, ResponseUpsert]):
```

Comment:
You can use generics to define this validator
```suggestion
import typing

R = typing.TypeVar("R", bound=BaseModel)
class ResponseValidator(typing.Generic[R]):
    def __init__(self, response_change: R):
```